File: backend-go/internal/company/controller/company_controller.go
package controller

import (
	"backend-go/internal/common/result"
	"backend-go/internal/company/model"
	"backend-go/internal/company/service"
	"strconv"

	"github.com/gofiber/fiber/v2"
)

type CompanyController struct {
	companyService *service.CompanyService
}

func NewCompanyController(companyService *service.CompanyService) *CompanyController {
	return &CompanyController{
		companyService: companyService,
	}
}

func (c *CompanyController) GetAll(ctx *fiber.Ctx) error {
	var param result.QueryParam
	if err := ctx.QueryParser(&param); err != nil {
		return err
	}
	res, err := c.companyService.FindEnabled(ctx.Context(), &param)
	if err != nil {
		return err
	}
	result.SetResult(ctx, res)
	return nil
}

func (c *CompanyController) Create(ctx *fiber.Ctx) error {
	var dto model.CompanyDTO
	if err := ctx.BodyParser(&dto); err != nil {
		return err
	}
	res, err := c.companyService.Create(ctx.Context(), &dto)
	if err != nil {
		return err
	}
	result.SetResult(ctx, res)
	return nil
}

func (c *CompanyController) Update(ctx *fiber.Ctx) error {
	var dto model.CompanyDTO
	if err := ctx.BodyParser(&dto); err != nil {
		return err
	}
	res, err := c.companyService.Update(ctx.Context(), &dto)
	if err != nil {
		return err
	}
	result.SetResult(ctx, res)
	return nil
}

func (c *CompanyController) Disable(ctx *fiber.Ctx) error {
	param := ctx.Params("id")
	id, err := strconv.Atoi(param)
	if err != nil {
		return err
	}
	err = c.companyService.Disable(ctx.Context(), id)
	if err != nil {
		return err
	}

	return nil
}

File: backend-java/src/main/java/pers/eastwind/billmanager/company/controller/CompanyController.java
package pers.eastwind.billmanager.company.controller;

import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.*;
import pers.eastwind.billmanager.common.model.PageResult;
import pers.eastwind.billmanager.common.model.QueryParam;
import pers.eastwind.billmanager.company.model.CompanyDTO;
import pers.eastwind.billmanager.company.service.CompanyService;

import java.util.List;

/**
 * 公司控制器
 */
@RestController
@RequestMapping("/api/company")
public class CompanyController {
    private final CompanyService companyService;

    public CompanyController(CompanyService companyService) {
        this.companyService = companyService;
    }

    @GetMapping
    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
    public PageResult<CompanyDTO> getAll(QueryParam queryParam) {
        return companyService.findEnabled(queryParam);
    }
    @PostMapping
    @PreAuthorize("hasAnyRole('ADMIN')")
    public CompanyDTO create(@RequestBody CompanyDTO companyDTO) {
        return companyService.create(companyDTO);
    }
    @PutMapping
    @PreAuthorize("hasAnyRole('ADMIN')")
    public CompanyDTO update(@RequestBody CompanyDTO companyDTO) {
        return companyService.update(companyDTO);
    }
    @DeleteMapping("/{id}")
    @PreAuthorize("hasAnyRole('ADMIN')")
    public void delete(@PathVariable Integer id) {
        companyService.disable(id);
    }
}
